### PR TITLE
Get/set agent config MAM func implemented (missing component configs) AEA-1070

### PR DESCRIPTION
**Pull request by panasevychol**
## Proposed changes

Get/set agent config MAM func implemented (missing component configs) AEA-1070

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by panasevychol** [comment]
Missing get_overridable_config for agent components because of lack of needed functionality in Config.
Described here: #1986

File: aea/manager.py [hunk 1]
```diff
@@ -577,6 +577,49 @@ def get_agent_alias(self, agent_name: str) -> AgentAlias:
...
+
+        :return: dict overridable config.
+        """
+        alias = self.get_agent_alias(agent_name)
+        config = alias.builder.loader.load_agent_config_from_json(alias.config)
+        agent_config = {
+            key: value
+            for key, value in config.json.items()
+            if key in config.FIELDS_ALLOWED_TO_UPDATE
+        }
```
Comment (DavidMinarsch) [inline aea/manager.py:594]: just extend `AgentConfig` to provide the overridable config in a property.

File: aea/manager.py [hunk 2]
```diff
@@ -577,6 +577,49 @@ def get_agent_alias(self, agent_name: str) -> AgentAlias:
...
+        # TODO: Get agent components overridable config the same way.
+        # we miss AgentConfig class functionality to do that.
```
Comment (DavidMinarsch) [inline aea/manager.py:596]: All the values in a component (excluding public id and type) are by definition overridable.

File: aea/manager.py [hunk 3]
```diff
@@ -664,9 +707,12 @@ def _make_config(
...
             components_configs[component_id] = obj
-        agent_update_dictionary["component_configurations"] = components_configs
-        # do the override (and valiation)
-        agent_config.update(agent_update_dictionary)
+
+        # update components
+        agent_config.update({"component_configurations": components_configs})
+        json_data = agent_config.ordered_json
+        json_data.update(agent_update_dictionary)
+        agent_config = AgentConfig.from_json(json_data)
```
Comment (DavidMinarsch) [inline aea/manager.py:715]: see my concern in other PR

File: aea/manager.py (hunk 2 above)
Comment (panasevychol) [reply aea/manager.py:596]: We need to have a property that lists all the fields except of that two as overridable.

File: aea/manager.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/manager.py:594]: something like `config.get_overridables() - > Dict:`

File: aea/manager.py [hunk 4]
```diff
@@ -577,6 +577,49 @@ def get_agent_alias(self, agent_name: str) -> AgentAlias:
...
+        alias = self.get_agent_alias(agent_name)
+        new_config = self._make_config(
+            alias.project.path,
+            agent_overrides=agent_overrides,
+            component_overrides=component_overrides,
+        )
+        new_alias = self._build_agent_alias(
+            alias.project, agent_name, config=new_config
+        )
+        self._agents[agent_name] = new_alias
```
Comment (DavidMinarsch) [inline aea/manager.py:621]:
```suggestion
        alias = self.get_agent_alias(agent_name)
        config = alias.builder.loader.load_agent_config_from_json(alias.config)
        config.update(
            agent_overrides=agent_overrides,
            component_overrides=component_overrides,
        )
        alias.config = config.to_json()
        self._agents[agent_name] = alias
```

Comment (DavidMinarsch) [reply aea/manager.py:621]: (roughly something like this)

File: aea/manager.py (hunk 3 above)
Comment (DavidMinarsch) [reply aea/manager.py:715]: (simplify)

**Comment by DavidMinarsch** [comment]
Closed in favour of #2079

**Pull request closed without merging**